tests/unit/test_kv_protocol.cpp

```diff
@@ -96,3 +96,57 @@ TEST_F(KiViProtocolTest, SendStringValueAndPassFailCount)
     std::string console = fake_console.get_stdout();
     ASSERT_EQ(console, output);
 }
+
+TEST_F(KiViProtocolTest, PerformsSetupHandshake)
+{
+    const int timeout = 99;
+    std::string test_name = "test";
+    std::string sync_msg = "{{" + std::string(GREENTEA_TEST_ENV_SYNC) + ";" + "0}}";
+    fake_console.set_stdin(sync_msg.c_str());
+
+    GREENTEA_SETUP(timeout, test_name.c_str());
+
+    std::string console = fake_console.get_stdout();
+    ASSERT_NE(console.find(sync_msg), std::string::npos);
+    ASSERT_NE(console.find(GREENTEA_CLIENT_VERSION_STRING), std::string::npos);
+    ASSERT_NE(console.find(std::to_string(timeout)), std::string::npos);
+    ASSERT_NE(console.find(test_name), std::string::npos);
+}
+
+TEST_F(KiViProtocolTest, SendsStartTestcaseMessage)
+{
+    std::string test_name = "test";
+
+    GREENTEA_TESTCASE_START(test_name.c_str());
+
+    std::string console = fake_console.get_stdout();
+    ASSERT_NE(console.find(GREENTEA_TEST_ENV_TESTCASE_START), std::string::npos);
+    ASSERT_NE(console.find(test_name), std::string::npos);
+}
+
+TEST_F(KiViProtocolTest, SendsFinishTestcaseMessage)
+{
+    std::string test_name = "test";
+    const int passes = 11;
+    const int failures = 99;
+
+    GREENTEA_TESTCASE_FINISH(test_name.c_str(), passes, failures);
+
+    std::string console = fake_console.get_stdout();
+    ASSERT_NE(console.find(GREENTEA_TEST_ENV_TESTCASE_FINISH), std::string::npos);
+    ASSERT_NE(console.find(test_name), std::string::npos);
+    ASSERT_NE(console.find(std::to_string(passes)), std::string::npos);
+    ASSERT_NE(console.find(std::to_string(failures)), std::string::npos);
+}
+
+TEST_F(KiViProtocolTest, SendsTestSuiteResultMessage)
+{
+    const int result = 1;
+
+    GREENTEA_TESTSUITE_RESULT(result);
+
+    std::string console = fake_console.get_stdout();
+    ASSERT_NE(console.find(GREENTEA_TEST_ENV_END), std::string::npos);
+    ASSERT_NE(console.find(GREENTEA_TEST_ENV_SUCCESS), std::string::npos);
+    ASSERT_NE(console.find(GREENTEA_TEST_ENV_EXIT), std::string::npos);
```

Comment:
For a lot of console output, doesn't order matter?

For example, shouldn't we fail the test if EXIT is sent before SUCCESS?

Reply:
Good point, will add assertions to ensure the order is correct.